scanpy/preprocessing/_simple.py

```diff
@@ -275,11 +276,16 @@ def log1p(
         Applies only to :class:`~anndata.AnnData`.
     chunk_size
         `n_obs` of the chunks to process the data in.
+    base
+        Base of the logarithm. Natural logarithm is used by default.
 
     Returns
     -------
     Returns or updates `data`, depending on `copy`.
     """
+    if 'log1p' in data.uns_keys():
+        logg.warning('adata.X seems to be already log-transformed.')
```

Comment:
nice!

Reply:
Do we want to allow something like: `sc.pp.log1p(adata, layer="logcounts")`? We do this with `normalize_total`, which you'd probably want to use before this.

Reply:
That’s for another feature request or PR, I’d say.

Reply:
I’d say that’s for another feature request or PR